pkg/operator/handler.go

```diff
@@ -125,12 +126,19 @@ func (h *Handler) syncDeploymentStatus(cr *regopapi.ImageRegistry, o runtime.Obj
 		logrus.Errorf("unable to update cluster status to %s=%s: %s", osapi.OperatorAvailable, osapi.ConditionTrue, errOp)
 	}
 
+	conditionRemoved(cr, resourceRemoved, operatorAvailableMsg, statusChanged)
```

Comment:
When we know that all resources have been removed, can we drop the finalizer?

Reply:
@dmage This code is wrong. Changed.